### PR TITLE
Bump stylelint fixes variables and switch from node-sass to sass

### PR DESCRIPTION
**Pull request by rodriciru**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Lint scss,remove unused scss vars, and bump stylelint to v13 (its required by stylelint-config-prestashop)
| Type?             |  improvement
| BC breaks?        |  no
| Deprecations?     |  no
| Fixed ticket?     | None.
| How to test?      | Compile and check everithing remains the same as before.
| Possible impacts? | None in my case

### CODE REVIEW

**Comment by SharakPL** [comment]
1. You've mixed it with your other PR about switching `node-sass` to `sass`. I think you're not supposed to do that.
2. Did you actually use this stylelint config? Afaik double quotes are required for strings.

**Comment by rodriciru** [comment]
Yep its missed, Sorry for the inconvenience if any.
Yes i use stylelint to fix errors and force a style rules based in stylelint-config-prestashop, which btw in his repo its more updated than in npm, but i stick to the npm package

Wait a moment... im 100% percent sure scss-fix put me single quotes on strings (which i hate btw). I run it again and put me double quotes. Commit and done. 
Yesterday I use a new set of rules for check for unused vars (https://github.com/primer/stylelint-config). Maybe it enforces single quotes. But relax, its removed after use, so i stick to 100% stylelint-config-prestashop rules only, no extra config.

**Comment by NeOMakinG** [comment]
> 1. You've mixed it with your other PR about switching `node-sass` to `sass`. I think you're not supposed to do that.
> 2. Did you actually use this stylelint config? Afaik double quotes are required for strings.

What's wrong with the fact that he's switching `node-sass` to `sass` in this PR, I would accept it as anyway we're pretty free for the moment on this project

It's pushing the project further and I don't see anything wrong atm 👍

**Comment by SharakPL** [comment]
I'd like to suggest making 2 files for variables: one for overriding bootstrap defaults and another for new custom variables. First one should enough to make whole custom theme without even adding single line of css. Second one could be loaded after whole bootstrap and provide setup for specific pages and components like `$product-name-color` or `$products-item-name-font-size`.

**Comment by NeOMakinG** [comment]
> I'd like to suggest making 2 files for variables: one for overriding bootstrap defaults and another for new custom variables. First one should enough to make whole custom theme without even adding single line of css. Second one could be loaded after whole bootstrap and provide setup for specific pages and components like `$product-name-color` or `$products-item-name-font-size`.

Agree with this, we already started it by splitting bootstrap variables into multiple files but `_variables.scss` should be cleaner 

@rodriciru would you mind splitting bootstrap variables from theme variables into 2 files? and change the import order

**Comment by rodriciru** [comment]
ok. i try

**Comment by NeOMakinG** [comment]
> ok. i try

Don't hesitate to come on the open source slack that you can find in the footer of https://www.prestashop-project.org/ if you want to speak or need help :)

**Comment by SharakPL** [comment]
> 
> 
> > 1. You've mixed it with your other PR about switching `node-sass` to `sass`. I think you're not supposed to do that.
> > 2. Did you actually use this stylelint config? Afaik double quotes are required for strings.
> 
> What's wrong with the fact that he's switching `node-sass` to `sass` in this PR, I would accept it as anyway we're pretty free for the moment on this project
> 
> It's pushing the project further and I don't see anything wrong atm 👍

Nothing wrong. Already said that switch to `sass` is a must since `node-sass` was depreciated. But there's already PR for that #120. Besides sass has different syntax than node-sass so theme scss changes should follow. Though I might have mixed myself a bit too because I checked this switch on classic theme instead of theme-refacto 😅 Problem is I can't even make docker to create container properly from Prestashop:develop ☹ Going to Slack soon...

**Comment by rodriciru** [comment]
i made a commit whit split bs vars etc.
One change that i made its that i dropped $grey-X colors by BS5 $gray-x00 colors, so maybe there its a little color changes, But i dont see any one.
BTW i replace some custom vars by BS5 Vars, and remove them if only used in one file (font family,plus some more)

**Comment by rodriciru** [comment]
Can cancel #120 and use this as a replacement ??? No need to merge 2 PR in one of them there its include the other changes, as @SharakPL says

**Review by NeOMakinG — APPROVED** [review]

**Comment by NeOMakinG** [comment]
Looks fine to me, gonna do some QA before merging anyway so please be patient 🙏

**Comment by rodriciru** [comment]
I see an error.
Someone wants to customize the template, for example he want to keep the BS5 default grey colors, just the colors. So they remove from \src\scss\partials\variables\bootstrap\_colors.scss, as they dont use.
This will throw an error on compile time, because \src\scss\partials\variables\bootstrap\_card.scss or whatever look for this var, wich doesent exist.
My solution is in theme.scss import all bootstrap, then our vars, and then again bootstrap.scss.
Everyone its happy.
Commit is on that way

**Comment by rodriciru** [comment]
Wait again please.
New problem. Scss maps are inmutable.
And BS5 uses maps a lot, so you have to know when a map is used and redefine yourself. This can be done, but i think its tedious.
New Proposal: Make a BS5 variables override file, where the use of references to BS5 vars are prohibited. Change the order of imports, first our var, then only the BS5 vars then the rest of our customizations, then the rest of BS5
Wdyt?

**Comment by SharakPL** [comment]
I recommend following [Bootstrap docs](https://getbootstrap.com/docs/5.1/customize/sass/#importing) with separating theme additions so theme.scss should look something like this:
```scss
@import "../node_modules/bootstrap/scss/functions";

@import "variables/bootstrapOverrides"; // only vars from "../node_modules/bootstrap/scss/variables"

@import "../node_modules/bootstrap/scss/variables";
@import "../node_modules/bootstrap/scss/mixins";
@import "../node_modules/bootstrap/scss/root";
@import "../node_modules/bootstrap/scss/utilities";
@import "../node_modules/bootstrap/scss/reboot";
@import "../node_modules/bootstrap/scss/type";
@import "../node_modules/bootstrap/scss/images";
@import "../node_modules/bootstrap/scss/containers";
@import "../node_modules/bootstrap/scss/grid";
@import "../node_modules/bootstrap/scss/helpers";

// 5. Optionally include utilities API last to generate classes based on the Sass map in `_utilities.scss`
@import "../node_modules/bootstrap/scss/utilities/api";

// Now we do theme with custom vars, mixins, plugins and styles
// that can't be done with Bootstrap's utility classes, mixins, functions, etc.
@import "variables/customFunctions";
@import "variables/customVariables";
@import "variables/customMixins";

@import "commons"; // now comes all the rest
@import "components/product";
@import "components/mainmenu";
...
```

**Comment by rodriciru** [comment]
> I recommend following [Bootstrap docs](https://getbootstrap.com/docs/5.1/customize/sass/#importing) with separating theme additions so theme.scss should look something like this:
> 
> ```css-scss
> @import "../node_modules/bootstrap/scss/functions";
> 
> @import "variables/bootstrapOverrides"; // only vars from "../node_modules/bootstrap/scss/variables"
> 
> @import "../node_modules/bootstrap/scss/variables";
> @import "../node_modules/bootstrap/scss/mixins";
> @import "../node_modules/bootstrap/scss/root";
> @import "../node_modules/bootstrap/scss/utilities";
> @import "../node_modules/bootstrap/scss/reboot";
> @import "../node_modules/bootstrap/scss/type";
> @import "../node_modules/bootstrap/scss/images";
> @import "../node_modules/bootstrap/scss/containers";
> @import "../node_modules/bootstrap/scss/grid";
> @import "../node_modules/bootstrap/scss/helpers";
> // and all the rest of BS
> 
> // 5. Optionally include utilities API last to generate classes based on the Sass map in `_utilities.scss`
> @import "../node_modules/bootstrap/scss/utilities/api";
> 
> // Now we do theme with custom vars, mixins, plugins and styles
> // that can't be done with Bootstrap's utility classes, mixins, functions, etc.
> @import "variables/customFunctions";
> @import "variables/customVariables";
> @import "variables/customMixins";
> 
> @import "commons"; // now comes all the rest
> @import "components/product";
> @import "components/mainmenu";
> ...
> ```

Fine, but a posible drawback is if Bootstrap will add a new file, for example _superCoolFeature.
If we update depencencies, we are not including this new file, this is why i just put the BS5 main file

**Comment by SharakPL** [comment]
[Option A](https://getbootstrap.com/docs/5.1/customize/sass/#importing) is also possible:

```scss
// Option A: Include all of Bootstrap

@import "variables/bootstrapOverrides"; // functions won't be available

@import "../node_modules/bootstrap/scss/bootstrap";  // this adds whole bootstrap, but maybe not all components are required...

// then all theme specifics
@import "variables/customFunctions";
@import "variables/customVariables";
@import "variables/customMixins";

@import "commons"; // now comes all the rest
@import "components/product";
@import "components/mainmenu";
...
```
That's just my recommendation. Don't know what maintainers would like here.

**Comment by rodriciru** [comment]
No problems here

**Comment by NeOMakinG** [comment]
> I recommend following [Bootstrap docs](https://getbootstrap.com/docs/5.1/customize/sass/#importing) with separating theme additions so theme.scss should look something like this:
> 
> ```css-scss
> @import "../node_modules/bootstrap/scss/functions";
> 
> @import "variables/bootstrapOverrides"; // only vars from "../node_modules/bootstrap/scss/variables"
> 
> @import "../node_modules/bootstrap/scss/variables";
> @import "../node_modules/bootstrap/scss/mixins";
> @import "../node_modules/bootstrap/scss/root";
> @import "../node_modules/bootstrap/scss/utilities";
> @import "../node_modules/bootstrap/scss/reboot";
> @import "../node_modules/bootstrap/scss/type";
> @import "../node_modules/bootstrap/scss/images";
> @import "../node_modules/bootstrap/scss/containers";
> @import "../node_modules/bootstrap/scss/grid";
> @import "../node_modules/bootstrap/scss/helpers";
> // and all the rest of BS
> 
> // 5. Optionally include utilities API last to generate classes based on the Sass map in `_utilities.scss`
> @import "../node_modules/bootstrap/scss/utilities/api";
> 
> // Now we do theme with custom vars, mixins, plugins and styles
> // that can't be done with Bootstrap's utility classes, mixins, functions, etc.
> @import "variables/customFunctions";
> @import "variables/customVariables";
> @import "variables/customMixins";
> 
> @import "commons"; // now comes all the rest
> @import "components/product";
> @import "components/mainmenu";
> ...
> ```

What's the benefit of this from your point of view?

**Comment by SharakPL** [comment]
The benefit is you can setup almost whole theme (everything BS cand handle) in just 1 variables file and all the theme specific additions in another. It will be easier to follow changes within Bootstrap and theme.

**Comment by SharakPL** [comment]
Isn't that obvious? Not all components may be needed in the theme, for example I would switch BS carousel for more robust and mobile friendly [Swiper](https://swiperjs.com/) that can be used anywhere on the site: mainslider, product carousels, thumbnail carousels, scrollable containers eg. for breadcrumbs.

**Comment by NeOMakinG** [comment]
> Isn't that obvious? Not all components may be needed in the theme, for example I would switch BS carousel for more versatile and mobile friendly [Swiper](https://swiperjs.com/) that can be used anywhere on the site: mainslider, product carousels, thumbnail carousels, scrollable containers eg. for breadcrumbs.

We had swiper before, but it's way bigger than BS carousel (which lack of feature of course), but I think we should provide a lightweight theme, if you want swiper, you should be able to add it

Anyway, maybe we could add a feature to switch between both

Of course it's obvious, but I wanted to have your advice on this, you don't need to be arrogant, Github is a rich discussion platform were we can gather some advice in order to make everyone happy.

**Comment by SharakPL** [comment]
Sorry, didn't mean to be arrogant, just confused about the question ;)

Swiper also can be customized. Only one animation type is fine + touch support and scrollable container support. I think nothing else would be needed.

**Comment by rodriciru** [comment]
i push a new commit. I try to remove some custom vars that are used few times, by BS5 vars.
Also i add !default flag to some of them, to allow delete whitout errors

**Comment by NeOMakinG** [comment]
> Sorry, didn't mean to be arrogant, just confused about the question ;)
> 
> Swiper also can be customized. Only one animation type is fine + touch support and scrollable container support. I think nothing else would be needed.

But I think I tested it with the lightest setup and it's pretty big, et should compare it with BS carousel and see if it's a good solution

File: src/scss/partials/_variables.scss
Comment (NeOMakinG) [inline src/scss/partials/_variables.scss:1]:
```suggestion
@import "variables/bootstrap/variable-overrides";
```

Keep naming consistent

**Review by NeOMakinG — COMMENTED** [review]
Just a little comment about naming but everything looks fine to me atm, we can continue the discussion outside about bootstrap variables, I'm not blocking this PR for this :)

**Review by NeOMakinG — APPROVED** [review]
Doing some QA on this after my meeting, then it's time to merge it if everything is fine :)

For the variable questions, let's open another PR if we decide something or an issue in order to discuss about it!

**Comment by NeOMakinG** [comment]
There are some strange behaviors linked to some colors you removed I think:
Before:
image

After:
image

(Search bar background color changed too, we should try to stay ISO with @prestascott mockups!

# Slider
Before:
image

After:
image


# Newsletter background is darker using the PR
Before:
image


Would you mind checking where every variables you used were used and try do be as much as possible near from the mockups? I think you can use bootstrap variables with some `lighten, darken` sass mixins if you don't want to add custom variables and avoid adding complexity :)

**Comment by SharakPL** [comment]
Might be related to that node-sass vs sass differences I was talking about earlier...

**Comment by rodriciru** [comment]
> Might be related to that node-sass vs sass differences I was talking about earlier...

Nope. Was my bad changin some gray-whatever to BS5 alternative gray-X00.
I think i fix all colors to make close to the original, but you tell me

**Comment by rodriciru** [comment]
the slider was bigger, but because it forces somo height and whid in the old version.
I remove this width and height ad put it in inline css (my bad). I will keep it like this, to improve Google CSL metrics.
If we have a slider whit diferent height images the page will jump, so not good.
Commit to remove inline css

**Comment by SharakPL** [comment]
> the slider was bigger, but because it forces somo height and whid in the old version. I remove this width and height ad put it in inline css (my bad). I will keep it like this, to improve Google CSL metrics. If we have a slider whit diferent height images the page will jump, so not good. Commit to remove inline css

IMO bootstrap carousel is just not good enough. Current Classic's setup (the same for previous versions) is awful. No wonder it's been always hidden by default on mobile :/ Maybe it would suffice for product images slider, where it's just a list of simple images of the same size, but not for main slider on the homepage with variable sizes of the slides and requirement for different rendering on mobile, especially in portrait mode.

**Comment by NeOMakinG** [comment]
> Might be related to that node-sass vs sass differences I was talking about earlier...

Looks like it's working fine tbh, only things I noticed were about changed variables

**Comment by NeOMakinG** [comment]
> > the slider was bigger, but because it forces somo height and whid in the old version. I remove this width and height ad put it in inline css (my bad). I will keep it like this, to improve Google CSL metrics. If we have a slider whit diferent height images the page will jump, so not good. Commit to remove inline css
> 
> IMO bootstrap carousel is just not good enough. Current Classic's setup (the same for previous versions) is awful. No wonder it's been always hidden by default on mobile :/ Maybe it would suffice for product images slider, where it's just a list of simple images of the same size, but not for main slider on the homepage with variable sizes of the slides and requirement for different rendering on mobile, especially in portrait mode.

Gonna do some tests with Swiper with the lightest way I can add it, if it's too heavy, I'll probably provide a solution where you can switch between both easily

**Review by NeOMakinG — APPROVED** [review]

**Comment by NeOMakinG** [comment]
I did import every bootstrap components partials instead of the whole bootstrap, overwise except the slider who need some works, everything looks fine to me, let's merge this one!

If one of you have something else to do with this PR, don't hesitate to open another one!

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**
